lib/improver/utilities/cube_units.py

```diff
@@ -141,6 +142,11 @@ def _find_dict_key(input_key):
         # this avoids duplicate results from key matching below
         return "probability"
 
+    exact_match = set([input_key]).intersection(DEFAULT_UNITS.keys())
+    if exact_match:
+        # returns the exact match if available
+        return list(exact_match)[0]
```

Comment:
I'm curious as to why this is needed.  The calling function `_get_required_units_and_dtypes` only calls this if it hasn't been able to find an exact match using `DEFAULT_UNITS[key]` (ie if this throws a `KeyError`).  What happens without this exact matching?

Reply:
Hmm, Good point. I remember when I wrote this that I thought it already did this somewhere, but I was having an issue with it complaining about having two matches. One was exact and one was partial. However, on reading the code now I can't see how that was happening. Maybe something was missing a single letter and I did not notice, but then my change would not have fixed it.

I have taken the change out and am re-running the tests. If they all pass I will just ditch it. Perhaps I was running the find_dict_key function in isolation for some reason.

Reply:
It all worked, so I will ditch this change.